### ISSUE TITLE
Output additional info for certificate

### ISSUE THREAD

**Issue by matteocorti** [matteocorti/nagios_plugins#100]
**Original reporter**: *anonymous*

Pls add an option to output additional information of certificate in Nagios. Example -O <which infos>
SSL_CERT OK - X.509 certificate for ... valid until Dec  2 21:31:09 2012 GMT

Issuer: Verisign (<country>)

Organisation: My Company (<department>)

City: New York (<region/county/state>)

Country: US

Email: test@abc.com

Alt.Names: test2.abc.com, test3.abc.com

Keyprint: 34 56 er 23

**Comment by matteocorti** [comment]
**Original reporter**: *[matteo@corti.li](https://github.com/matteo@corti.li)*

Hi,

could be a nice idea. Just to be sure you mean to add an option to be able to configure what is printed in the status?

For example if I say:

`-O Country,Keyprint`the plugin will print

`SSL_CERT OK - X.509 certificate for localhost from USA with keyprint 34 56 er 23 valid until Dec 2 21:31:09 2012 GMT`?

**Comment by matteocorti** [comment]
**Original reporter**: *[matteo@corti.li](https://github.com/matteo@corti.li)*

The issue is partially resolved with commit r1303. I added the possibility to specify a list of additional attributes that have to printed in the plugin  long output (LONGSERVICEOUTPUT). The list is limited to the attributes that can be delivered by openssl with direct request.

I decided not to parse the output to query for additional parameters since it would make a clean and safe implementation difficult.
